### PR TITLE
discovery: report containerID for containerised services

### PR DESCRIPTION
**Pull request by Yumasi**
### What does this PR do?

This PR makes the Discovery agent check add container IDs to discovered services. It relies on the workload metadata store to get the container IDs.

### Motivation

[USMON-1204](https://datadoghq.atlassian.net/browse/USMON-1204?atlOrigin=eyJpIjoiZjUzNDc0MWVjNTc1NGQxY2E0NGUzOTBmMTczOTU2OTAiLCJwIjoiaiJ9)

### Additional Notes

[USMON-1204]: https://datadoghq.atlassian.net/browse/USMON-1204?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by vickenty — APPROVED** [review]

**Review by vitkyrka — COMMENTED** [review]
Do you think it would be possible to test this in the E2E test (can be done in a separate PR)?

File: pkg/collector/corechecks/servicediscovery/impl_linux.go [hunk 1]
```diff
@@ -100,6 +110,12 @@ func (li *linuxImpl) DiscoverServices() (*discoveredServices, error) {
 				li.ignoreProcs[pid] = true
 				continue
 			}
+
+			if id, ok := containers[pid]; ok {
+				svc.service.ContainerID = id
```
[the remark this answers is not in the document]
Comment (Yumasi) [reply pkg/collector/corechecks/servicediscovery/impl_linux.go]: Made a new method for handling the `potential->alive` case, and moved the addition of the container id in it.

**Review by vitkyrka — APPROVED** [review]

File: pkg/collector/corechecks/servicediscovery/cp_stub.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+type containerProviderStub struct {
+	pidToCid map[int]string
+}
+
+func newContainerProviderStub(targetPIDs []int) proccontainers.ContainerProvider {
+	pidToCid := make(map[int]string)
+
+	for _, pid := range targetPIDs {
+		pidToCid[pid] = "abcd"
+	}
```
Comment (guyarb) [inline pkg/collector/corechecks/servicediscovery/cp_stub.go:27]: what is `abcd`?

If that's only a testutility, then please add `test` build flag

Comment (Yumasi) [reply pkg/collector/corechecks/servicediscovery/cp_stub.go]: Added the build flag.

> what is abcd?
Just a dummy Container ID used in the test. I moved it to a const value to sync tests and stub code and prevent mistakes.

**Pull request merged**